torch/distributed/device_mesh.py

```diff
@@ -222,9 +224,9 @@ def __init__(
             self._flatten_mesh_list = tuple(self.mesh.flatten().tolist())
             self._hash = hash((self._flatten_mesh_list, self.mesh.shape, id(self)))
 
-            # Skip process group initialization if xla device.
+            # Skip process group initialization if xla device or init backend is False
             # TODO(yeounoh) implement DeviceMesh backend and register XLA backend.
-            if device_type != "xla":
+            if device_type != "xla" and _init_backend:
```

Comment:
If we skip `_get_or_create_default_group()`, then `self._coordinate_on_dim` will not be defined when `_init_backend=False`.

Reply:
oh I think we should fix this it seems, I'll do it as a follow up PR

Reply:
I think I did similar thing for nD submesh slicing and not initing the sub pgs again for the mesh slice won't work as there is dim_group_infos for the mesh slice. iirc, that's why I need to revert.

Reply:
Yes, upon more investigation, I think we need to populate both `self._coordinate_on_dim` and `self._dim_group_infos`, so this current change should break unit tests IIUC.

Reply:
It won't break any test yet as _init_backend is default to True, but a followup is needed for most of the API to work with _init_backend = False.

Reply:
I think we would always want the default `_init_backend` to be True, only when we do submesh slicing we would manually turn it off